src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -186,10 +187,43 @@ public void start(final Map<String, String> props) {
         Optional<String> jsonDocument = Optional.empty();
         if (publishFullDocumentOnly) {
           if (changeStreamDocument.containsKey("fullDocument")) {
-            jsonDocument = Optional.of(changeStreamDocument.getDocument("fullDocument").toJson());
+            if (sourceConfig.getJsonType().equals("canonical")) {
+              jsonDocument = Optional.of(changeStreamDocument.getDocument("fullDocument").toJson());
+            } else if (sourceConfig.getJsonType().equals("relaxed")) {
+              jsonDocument = Optional.of(changeStreamDocument.getDocument("fullDocument").toJson(JsonWriterSettings.builder()
+                      .outputMode(JsonMode.RELAXED)
```

Comment:
Nope it's not ! Here the default value is `JsonMode.canonical` because it's the .parseJson() used before and the output was : 
`"timestamp":{"$long":"value"}"` by adding this line all output was like `"timestamp":value`"

Reply:
The documentation is correct. In version 3.12.x, [`BsonDocument.toJson()`](https://github.com/mongodb/mongo-java-driver/blob/3.12.x/bson/src/main/org/bson/BsonDocument.java#L824) creates his own `JsonWriterSettings` using the default values, of which [`outputMode` is `JsonMode.STRICT`](https://github.com/mongodb/mongo-java-driver/blob/3.12.x/bson/src/main/org/bson/json/JsonWriterSettings.java#L127), whereas `JsonWriterSettings.builder()` will return a new builder where [the default value for `outputMode` is `Json.RELAXED`](https://github.com/mongodb/mongo-java-driver/blob/3.12.x/bson/src/main/org/bson/json/JsonWriterSettings.java#L565).

Reply:
Note that in version 4.x, [the default value has been changed to `JsonMode.RELAXED`](https://github.com/mongodb/mongo-java-driver/blob/4.0.x/bson/src/main/org/bson/BsonDocument.java#L826).

Reply:
So here to close this issue the Mongo Driver use by Mongo Source is in `3.12` so it's mean that the `STRICT` mode is the legacy